src/process/executable_path.cr

```diff
@@ -28,13 +28,42 @@ class Process
     end
   end
 
+  private def self.is_executable_file?(path)
+    unless File.info?(path, follow_symlinks: true).try &.file?
+      return false
+    end
+    {% if flag?(:win32) %}
+      true
+    {% else %}
+      File.executable?(path)
+    {% end %}
+  end
+
   # Searches an executable, checking for an absolute path, a path relative to
   # *pwd* or absolute path, then eventually searching in directories declared
   # in *path*.
   def self.find_executable(name : Path | String, path : String? = ENV["PATH"]?, pwd : Path | String = Dir.current) : String?
-    name = Path.new(name)
+    find_executable_possibilities(Path.new(name), path, pwd) do |p|
+      if is_executable_file?(p)
+        return p.to_s
+      end
+    end
+    nil
+  end
+
+  private def self.find_executable_possibilities(name, path, pwd)
+    return if "#{name}".empty?
+
+    {% if flag?(:win32) %}
+      basename = name.ends_with_separator? ? "" : name.basename
```

Comment:
This would be greatly improved by a comment describing the reason and result of this special handling of win32 paths.

Reply:
To match specs, basically! So I'll add just that probably